KNOWLEDGE.md

```
@@ -4108,3 +4108,46 @@ if (llvm::Type *mapKeyTy = getMapKeyType(s)) {
+**How to apply**: For each overload candidate, track a
+`std::vector<bool> needsWidening` alongside the existing `paramLLVMTypes`
+vector. Before emitting `CreateCall`, loop over the matched args and call
+`emitWideningConversion` wherever `needsWidening[i]` is true. The error path
+for "no matching overload" is unchanged — falling through Pass 1 and Pass 2
+both lands in the original `"argument N requires <type>"` message.
+
+**Out of scope / known gap**: Math custom emitters
+(`emitMathFloorCeilRound`, `emitMathLog`, `emitMathAbs`, `emitMathPow` for
+mixed-type calls) have their own hardcoded `!= cg.f64Ty_` guards and bypass
+the table-driven path entirely. They still reject `int` args. Tracked in
+#1230.
```

markdownlint-cli2 (0.22.0)

[warning] No space after hash on atx style heading

(MD018, no-missing-space-atx)